tests/zfs-tests/tests/functional/tmpfile/setup.ksh

```diff
@@ -31,9 +31,12 @@
 
 . $STF_SUITE/include/libtest.shlib
 
-if ! $STF_SUITE/tests/functional/tmpfile/tmpfile_test /tmp; then
+DISK=${DISKS%% *}
+default_setup_noexit $DISK
+
+if ! $STF_SUITE/tests/functional/tmpfile/tmpfile_test $TESTDIR/tmpfile; then
```

Comment:
Is `$TESTDIR/tmpfile` a file or directory?
O_TMPFILE should take a directory as pathname argument.